tests/vibe.web.rest.1140/source/app.d

```diff
@@ -0,0 +1,52 @@
+import std.datetime;
+import vibe.d;
+
+interface IOrientDBRoot
+{
+	@property IOrientDBQuery query();
+}
+
+interface IOrientDBQuery
+{
+	@method(HTTPMethod.GET)
+	@path(":db_name/sql/:query/:result_set_size")
+	Json sql(string _db_name, string _query, int _result_set_size);
+}
+
+class OrientDBRoot : IOrientDBRoot {
+	private OrientDBQuery m_query;
+	override @property IOrientDBQuery query() { return m_query; }
+	public this() { this.m_query = new OrientDBQuery(); }
+}
+
+class OrientDBQuery : IOrientDBQuery {
+	override Json sql(string _db_name, string _query, int _result_set_size) {
+		assert(_db_name == Param1, _db_name);
+		assert(_query == Param2, _query);
+		assert(_result_set_size == Param3, to!string(_result_set_size));
+		return Json.emptyObject;
+	}
+}
+
+enum Param1 = "twitter_data";
+enum Param2 = "select DownloadedDateTime from Message";
```

Comment:
@Geod24 I thought on the issue and I find the following:
Lets for the time forget about REST and solve a simple task.
I want to create an `/sql/{urlEncode(select date from files where path = '/')}/1` query.
It should looks like `/sql/select%20date%20from%20files%20where%20path%20%3D%20%27%2F%27/1`.
Is this task legit? I think so.
Now lets try to solve it.
The first try:
```
auto url = URL("http", db_host, db_port, Path("/sql/select date from files where path = '/'/1"));
url.username = db_username;
url.password = db_password;
auto req = requestHTTP(url);
```
Obviously, fail: `Path` splits the url path by '/'.
Ok. Lets make try two:
```
auto url = URL("http", db_host, db_port, Path("/sql/select%20date%20from%20files%20where%20path%20%3D%20%27%2F%27/1"));
url.username = db_username;
url.password = db_password;
auto req = requestHTTP(url);
``` 
Strange result: `url.localURI` returns `/sql/select%20date%20from%20files%20where%20path%20%3D%20%27/%27/1`: All ok, but `%2f` has been decoded back to `/`
```
url.localURI = "/sql/select%20date%20from%20files%20where%20path%20%3D%20%27%2F%27/1";
```
same result.